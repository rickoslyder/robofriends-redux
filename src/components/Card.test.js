import { render, screen } from '@testing-library/react';
import React from 'react';
import Card from './Card';

const mockCardComponent = <Card name={"test"} email={"test@test.com"} id={69} />

test('expect to render Card component', () => {
    render(mockCardComponent)
    const header = screen.getAllByRole("heading")
    expect(header).toBeDefined();
    expect(header.length).toEqual(1)
})

test('expect mock Card component JSX to match snapshot', () => {
    expect(mockCardComponent).toMatchSnapshot()
})

test('expect rendered mock Card component to match snapshot', () => {
    expect(render(mockCardComponent).baseElement).toMatchSnapshot()
})