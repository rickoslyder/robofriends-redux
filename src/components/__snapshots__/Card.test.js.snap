// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect mock Card component JSX to match snapshot 1`] = `
<Card
  email="test@test.com"
  id={69}
  name="test"
/>
`;

exports[`expect rendered mock Card component to match snapshot 1`] = `
<body>
  <div>
    <div
      class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
    >
      <img
        alt="robots"
        src="https://robohash.org/69?size=200x200"
      />
      <div>
        <h2>
          test
        </h2>
        <p>
          test@test.com
        </p>
      </div>
    </div>
  </div>
</body>
`;
